third_party/move/move-model/src/ty.rs

```diff
@@ -543,6 +544,32 @@ impl Constraint {
                 *a1 = a1.union(*a2);
                 Ok(true)
             },
+            // After the above checks on same type of contraint
+            // Check compatibility between ability and number
```

Comment:
Are there other combinations to consider, e.g., references etc?

Could you add a comment about why this particular combination requires the special check?

Reply:
Added the comment